src/web/src/app/space/add/component.ts

```diff
@@ -0,0 +1,123 @@
+import { startWith, map, debounceTime, distinctUntilChanged, switchMap } from 'rxjs/operators';
+import { MatOption } from '@angular/material/core';
+import { Router, NavigationExtras } from '@angular/router';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+
+import { SpaceService } from '../service';
+import { Space } from '../model';
+import { SpaceKeyValidation } from './space-key-validator';
+
+@Component({
+	selector: 'app-space-add',
+	templateUrl: './component.html',
+	styleUrls: ['./component.css'],
+})
+export class SpaceAddComponent implements OnInit {
+	createForm: FormGroup;
+	controlAutocomplete = new FormControl('');
+
+	spaces$!: Observable<Space[]>;
+	selectedSpace: Space;
+	keyWasChanged: boolean = false;
+
+	constructor(
+		private router: Router,
+		private fb: FormBuilder,
+		private space: SpaceService,
+		private spaceKeyValidator: SpaceKeyValidation,
+	) {
+		this._createForm();
+	}
+
+	ngOnInit() {
+		this.spaces$ = this.controlAutocomplete.valueChanges.pipe(
+			startWith(''),
+			debounceTime(300), // Optional: debounce input changes to avoid excessive requests
+			distinctUntilChanged(), // Optional: ensure distinct values before making requests
+			switchMap(value => {
+				if (!value)
+					this.selectedSpace = null;
+				return this.space.getList(10, 0, value || '').pipe(
+					map(res => res.results)
+				);
+			})
+		);
+	}
+
+	onSelectOption(option: MatOption) {
+		if (option) {
+			this.selectedSpace = {
+				id: option.value.id,
+				name: option.value.name,
+				key: option.value.key,
+				requestsAllowed: option.value.requestAllowed,
+				createdAt: option.value.createdAt
+			}
+		}
+	}
+
+	sendRequestToJoin() {
+		const navigationExtras: NavigationExtras = {state: {spaceName: this.selectedSpace.name}};
+		this.router.navigate(['space/add/request'], navigationExtras);
+	}
+
+	private translitFromRuToEn = {
+		"ё": "yo", "й": "i", "ц": "ts", "у": "u", "к": "k", "е": "e", "н": "n", "г": "g",
+		"ш": "sh", "щ": "sch", "з": "z", "х": "h", "ъ": "'", "ф": "f", "ы": "i", "в": "v", "а": "a", "п": "p", "р": "r", "о": "o", "л": "l", "д": "d", "ж": "zh", "э": "e", "я": "ya", "ч": "ch", "с": "s", "м": "m", "и": "i", "т": "t", "ь": "'", "б": "b",
+		"ю": "yu"
+	};
+
+	private _transliterate(word: string): string {
+		return word
+			.split('')
+			.map(char => this.translitFromRuToEn[char] || char)
+			.join('');
+	}
+
+	onNameChange(name: string) {
+		if (!this.keyWasChanged) {
+			let newKey = name.toLowerCase();
+			newKey = this._transliterate(newKey);
+			newKey = newKey.replace(/[^a-z0-9_]/g, '');
+			this.createForm.patchValue({
+				key: newKey
+			});
+		}
+	}
+
+	onKeyChange(key: string) {
+		this.keyWasChanged = true;
+	}
+
+	onSubmitCreate(data): void {
+		if (this.createForm.invalid) {
+			return;
+		}
+		this.space.createNew(data.name, data.key, data.requestsAllowed)
+			.subscribe(res => { 
+				if (res == true){
+					this.router.navigate(['space/list']);
+				}
+			});
```

Comment:
1. А в каких случаях тут res может быть false ?
2. C true сравнивать не обязательно. Лучше просто if (res)
```suggestion
			.subscribe(_ => this.router.navigate(['space/list']));
```

Reply:
Готово.